### PR TITLE
PR: Docs issue #52

### PR DESCRIPTION
**Pull request by nelsonic**
+ [x] Adding docs for #52 ... 👨‍💻 https://hexdocs.pm/fields/2.1.0/api-reference.html

The difference is subtle:
image

But something as basic as adding syntax highlighting and explanatory comments makes a _big_ difference:
image

**`before`**: [https://github.com/dwyl/fields/](https://github.com/dwyl/fields/tree/43f15171df363fa7f71ce14e36f44be41916b332)
**`after`**: https://github.com/dwyl/fields/tree/update-docs-issue-%2352
+ [x] Instructions for required environment variables #36 

I also went through the backlog and reviewed all the open issues, closed most and implemented these:
+ [x] Implements `:name` Binary Field #11 
+ [x] Adds `Fields.UrlEncrypted` for #16 
+ [x] Add New Callback functions required by Ecto 3.0 #60

### CODE REVIEW

**Comment by nelsonic** [comment]
Hi @SimonLab, 👋 
Hope you had a good weekend. ☀️ 
I've attempted to improve the docs in this project to make it more beginner-friendly. 💭 
+ **`before`**: [https://github.com/dwyl/fields/](https://github.com/dwyl/fields/tree/43f15171df363fa7f71ce14e36f44be41916b332)
+ **`after`**: https://github.com/dwyl/fields/tree/update-docs-issue-%2352

Please take a look and if you have any thoughts/comments, please share. 📝 
Thanks. ✨

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # Fields
 
-A collection of commonly used fields implemented as custom Ecto types 
+A collection of commonly used fields implemented as custom Ecto types
 with the validation, sanitising and encryption/hashing. <br />
```
Comment (SimonLab) [inline README.md]: I'm not sure but the `with the` sounds a bit weird to me, should it just
```suggestion
with validation, sanitising and encryption/hashing. <br />
```
but maybe it's fine

File: README.md [hunk 2]
```diff
@@ -11,149 +11,216 @@ with built-in validation, testing, sanitising and encryption.
...
-## Background / Further Reading
+## 3. Apply the relevant field(s) to your schema 📝
 
+Each field can be used in place of an Ecto type when defining your schema.
 
-### How to Create a Re-useable Elixir Package and Publish to `hex.pm`
-+ https://hex.pm/docs/publish
-+ https://medium.com/kkempin/how-to-create-and-publish-hex-pm-package-elixir-90cb33e2592d
-+ https://medium.com/blackode/how-to-write-elixir-packages-and-publish-to-hex-pm-8723038ebe76
+An example fo defining a "user" schema using **Fields**:
```
Comment (SimonLab) [inline README.md]:
```suggestion
An example for defining a "user" schema using **Fields**:
```

File: README.md (hunk 1 above)
Comment (nelsonic) [reply README.md]: Good catch. it's bad grammar. 👍

File: README.md (hunk 2 above)
Comment (nelsonic) [reply README.md]: Well spotted. 👍 🙄 😄

File: README.md [hunk 3]
```diff
@@ -11,149 +11,216 @@ with built-in validation, testing, sanitising and encryption.
...
-  field(:postcode, Fields.Postcode)
-  field(:password, Fields.Password)
+In order to use Encryption and Hashing,
+you will need to have environment variables
+defined for `ENCRYPTION_KEYS` and `SECRET_KEY_BASE` respectively.
 
-  timestamps()
-end
+```yaml
+export ENCRYPTION_KEYS='nMdayQpR0aoasLaq1g94FLba=,L+ZVX8iheoqgqb22mUpATmMDsvVGtafoAeb='
```
Comment (SimonLab) [inline README.md]: I'm wondering if we need to have two keys in the `ENCRYPTION_KEYS` or if we should just have one to make the example simple?
Or we could add a link  to https://github.com/dwyl/phoenix-ecto-encryption-example#2-create-the-user-schema-database-table which mention key rotation?

File: README.md [hunk 4]
```diff
@@ -2,158 +2,225 @@
...
+the `load/1` callback will be called,
+giving you the data in the format you need.
+`Fields.EmailEncrypted` will be decrypted back to plaintext.
+This all happens 100% transparently to the developer.
+It's _like_ magic. But the kind where you can
+actually _understand_ how it works!
+(_if you're curious, read the
+[**`code`**](https://github.com/dwyl/fields/tree/master/lib)_)
+
+Each Field optionally defines an `input_type/0` function.
```
Comment (SimonLab) [inline README.md]: At the moment only the `HtmlBody` and `DescriptionPlaintextUnlimited` fields define this function. We might want to add it to the other fields

Comment (nelsonic) [reply README.md]: @SimonLab absolutely. We need to add this to the other fields ASAP. 👍 
Would you mind creating an issue with a checklist for all the remaining fields that don't yet have it?

File: README.md (hunk 3 above)
Comment (nelsonic) [reply README.md]: @SimonLab good point. we only need _one_ value in `ENCRYPTION_KEYS` for simplicity.

Comment (SimonLab) [reply README.md]: The Background / Further Reading section mention already the phoenix-ecto-encryption-example
image

Comment (nelsonic) [reply README.md]: I've updated the `.env_sample` and `README.md` to only have one Encryption Key. 👍

**Pull request merged**